### PR TITLE
test: Improve/fix handling browser error messages from our own pages

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

**Comment by martinpitt** [comment]
First round: networking and storage are happy. /other failures are reasonably consistent, and the [fedora-38](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19506-20231019-102429-9f6eda28-fedora-38-other/log.html) run is a good representative. Likewise, for /expensive, the 
[f38 run](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19506-20231019-102429-9f6eda28-fedora-38-expensive/log.html) is representative, other OSes don't fail anything else.

File: pkg/shell/machines/machines.js [hunk 1]
```diff
@@ -220,15 +220,15 @@ function Machines() {
...
-        const bridge = cockpit.dbus(null, { bus: "internal", superuser: "try" });
+        const bridge = cockpit.dbus(null, { bus: "internal", superuser: "require" });
         const mod =
             bridge.call("/machines", "cockpit.Machines", "Update", ["99-webui.json", host, values_variant])
-                    .catch(function(error) {
-                        console.error("failed to call cockpit.Machines.Update(): ", error);
+                    .catch(error => {
+                        // avoid make noise when we are not superuser
+                        if (error.problem !== "access-denied")
+                            console.error("failed to call cockpit.Machines.Update(): ", JSON.stringify(error));
```
Comment (cockpituous) [inline pkg/shell/machines/machines.js:229]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19506-20231019-123114-02c8ffd1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (martinpitt) [reply pkg/shell/machines/machines.js:229]: Right, it's a "Should Not Happen™" now.

**Review by jelly — APPROVED** [review]
Great cleanups! :fireworks:

**Pull request merged**